### DISCUSSION TITLE
How to limit the size of a React Flow Diagram.

### DISCUSSION THREAD

**Discussion by jbrighton-dev** [xyflow/xyflow#3530] [General]
What I'm trying to achieve is to get a React Flow diagram in a box on a webpage, where it doesn't take up the entire screen. However, all my experience so far is that it doesn't play nicely with other elements and that the css must be height 100%, width 100% or it doesn't work. Looking at the below example, how would I align the text to the right and left, above and below the diagram box. 
![image](https://github.com/wbkd/react-flow/assets/17567353/6ae9ef75-103b-43a0-85ba-e76cd47a18f3)

**Comment by bcakmakoglu** [comment]
I think this should be achievable using flex-boxes.

Something like this

```jsx
export default function App() {
  return (
    <div
      className="App"
      style={{
        display: "flex",
        gap: "8px"
      }}
    >
      <div>Text</div>

      <div
        style={{
          display: "flex",
          flexDirection: "column",
          gap: "8px",
          height: "100%",
          width: "100%"
        }}
      >
        Text
        <div
          style={{
            height: "100%",
            backgroundColor: "gray"
          }}
        >
          <Flow />
        </div>
        Text
      </div>

      <div>Text</div>
    </div>
  );
}
```

[Here's an example.](https://codesandbox.io/s/elegant-villani-dx2mw9?file=/src/App.tsx:52-637)
